pmd-core/src/main/java/net/sourceforge/pmd/RuleBuilder.java

```diff
@@ -0,0 +1,246 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.w3c.dom.Element;
+
+import net.sourceforge.pmd.lang.Language;
+import net.sourceforge.pmd.lang.LanguageRegistry;
+import net.sourceforge.pmd.lang.LanguageVersion;
+import net.sourceforge.pmd.properties.PropertyDescriptor;
+
+
+/**
+ * Builds a rule, validating its parameters throughout. The builder can define property descriptors, but not override
+ * them. For that, use {@link RuleFactory#decorateRule(Rule, Element)}.
+ *
+ * @author Clément Fournier
+ * @since 6.0.0
+ */
+class RuleBuilder {
```

Comment:
public?

Reply:
There isn't much of a use case outside of RuleFactory, so I figured it could stay hidden... But if I move it to another package I'll have to make it public